src/core/data/grid/gridlayoutdefs.h

```diff
@@ -74,9 +112,9 @@ namespace core
         static constexpr QtyCentering primal = QtyCentering::primal;
         static constexpr QtyCentering dual   = QtyCentering::dual;
 
-        static constexpr uint32 idirX = static_cast<uint32>(Direction::X);
-        static constexpr uint32 idirY = static_cast<uint32>(Direction::Y);
-        static constexpr uint32 idirZ = static_cast<uint32>(Direction::Z);
+        static constexpr uint32 idirX = Direction::X;
```

Comment:
having "std::" before fixed width integers is largely superfluous I think, but it's mostly "uint32" so I'll use that 
there's a question around whether these types should be configurable at compile time in case it's too small but that's for another day

Reply:
there's a number of places with "uin32_t" already it seems

Reply:
> there's a number of places with "uin32_t" already it seems

we should have a uniform usage across the code base... let's use the alias

Reply:
a task for the int refactoring PR perhaps